Lib/logging/__init__.py

```diff
@@ -1543,9 +1558,17 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        try:
+            return self._cache[level]
+        except KeyError:
+            _acquireLock()
+            if self.manager.disable >= level:
```

Comment:
@vsajip The only thing I see is `LoggerAdapter.getEffectiveLevel()` could be overwritten and that would change the behavior of `LoggerAdapter.isEnabledFor()`. That would not be the case if `LoggerAdapter.isEnabledFor()` pointed to `Logger.isEnabledFor()`. So if we want caching on adapters (and we probably do), the caching logic should be duplicated on the adapter. The alternative is to give up some performance and move caching to `getEffectiveLevel()`. Thoughts?